rsp_jupyter_extensions/handlers/tutorials.py

```diff
@@ -382,5 +290,127 @@ def post(self) -> None:
+            return Hierarchy()
+        repo_url, branch = repo.split("@")
+        if not branch:
+            # This is to placate mypy: _find_repo() will append @main
+            # if needed
+            branch = "main"
+        if not from_cache:
+            self.log.debug("get_gh: New clone")
+            self.log.debug(f"Cloning {repo_url} branch {branch} to {dirname}")
+            _clone_repo(repo_url, branch, dirname)
+        dir_obj = Path(dirname)
+
+        def _xform_src(src: Path | str) -> str:
+            # From our cloned copy, reverse-engineer the direct download
+            # file URLs for each notebook.
+            if isinstance(src, str):
+                src = Path(src)
+            rest = src.relative_to(dir_obj)
+            return urlunparse(
+                urlparse(f"{repo_url}/raw/refs/heads/{branch}/{rest!s}")
+            )
+
+        def _xform(src: Path) -> Path:
+            homedir = _get_homedir()
+            rest = src.relative_to(dir_obj)
+            return (tutorial_dir / rest).relative_to(Path(homedir))
+
+        return self._build_hierarchy(
+            dir_obj,
+            parent=Path("/"),
+            suffix=".ipynb",
+            action=Actions.FETCH,
+            xform_src=_xform_src,
+            xform_dest=_xform,
+        )
+
+    def _build_hierarchy(
+        self,
+        root: Path,
+        parent: Path = Path("/"),
+        action: Actions = Actions.COPY,
+        disposition: Dispositions = Dispositions.PROMPT,
+        xform_src: Callable[[str | Path], str | Path] = lambda x: x,
+        xform_dest: Callable[[Path], Path] = lambda x: x,
+        suffix: str | None = None,
+    ) -> Hierarchy:
+        h = Hierarchy()
+        sortlist = list(root.iterdir())
+        sortlist.sort(key=lambda x: x.name)
+        for entry in sortlist:
+            self.log.debug(f"build_hierarchy: considering {entry.name}")
+            # We may want to make this more sophisticated sometime.
+            if entry.is_symlink():
+                # Just skip any symbolic links.
+                self.log.debug(f"Skipping symlink {entry.name}")
+                continue
+            if entry.is_dir():
+                if entry.name == ".git":
+                    # Skip the Git objects dir: the tutorials hierarchy is
+                    # not intended for pushing back up (consider that
+                    # the repository tags probably do not match the eups
+                    # container tag)
+                    self.log.warning("build_herarchy: Skipping .git directory")
+                    continue
+                next_parent = parent / entry.name
+                # Recurse down the tree.
+                self.log.debug(f"build_hierarchy: recurse on {next_parent!s}")
+                subdir = self._build_hierarchy(
+                    root=entry,
+                    parent=next_parent,
+                    action=action,
+                    disposition=Dispositions.PROMPT,
```

Comment:
Minor detail but since this matches the default value, perhaps cleaner to use the param `disposition`?

Reply:
Do you mean, omit it and let the default handle it?

Reply:
OK.